CONTRIBUTING.md

```diff
@@ -576,6 +579,61 @@ that are part of this project but not production code. The canonical example
 of this is `junit`.</dd>
 </dl>
 
+
+Reviewing and accepting your contribution
+-----------------------------------------
+
+We review every contribution carefully to ensure that the change is of high
+quality and fits well with the rest of the Elasticsearch codebase. If accepted,
+we will merge your change and usually take care of backporting it to
+appropriate branches ourselves.
+
+We really appreciate everyone who is interested in contributing to
+Elasticsearch and regret that we sometimes have to reject contributions even
+when they might appear to make genuine improvements to the system. Reviewing
+contributions can be a very time-consuming task, yet the team is small and our
+time is very limited. In some cases the time we would need to spend on reviews
+would outweigh the benefits of a change by preventing us from working on other
+more beneficial tasks instead.
+
+Please discuss your change in a Github issue before spending much time on its
+implementation. We sometimes have to reject contributions that duplicate other
+efforts, take the wrong approach to solving a problem, or solve a problem which
+does not need solving. An up-front discussion would have saved a good deal of
+wasted time in these cases.
+
+We normally immediately reject isolated PRs that only perform simple
+refactorings or otherwise "tidy up" certain aspects of the code. We think the
+benefits of this kind of change are very small, and in our experience it is not
+worth investing the substantial effort needed to review them. This especially
+includes changes suggested by tools.
+
+We sometimes reject contributions due to the low quality of the submission
+since low-quality submissions tend to take unreasonable effort to review
+properly. Quality is rather subjective so it is hard to describe exactly how to
+avoid this, but there are some basic steps you can take to reduce the chances
+of rejection. Follow the guidelines listed above when preparing your changes.
+You should add tests that correspond with your changes, and your PR should pass
+the full test suite too. It makes it much easier to review if your code is
```

Comment:
Community PRs don't automatically hit CI, so it's hard for a contributor to know if their PR passes the full test suite without someone with CI privileges getting involved to kick off a run.  Personally, I think it's sufficient to say a change should pass `./gradlew test` and `./gradlew precommit`  before asking for a review, but if stuff fails in CI during the review process, that's not a reason for rejection IMHO.

Reply:
Hmm. The preceding docs in this file (and in `TESTING.md`) describe how to run the tests in much more detail than I think we should cover here. Changes _should_ pass the full test suite, and I do think we should say that we expect folks to try and verify that themselves. Would `s/full test suite/affected test suites/` be satisfactory? If external contributors are struggling to run the relevant test suites then that sounds like a bug.

Of course this is a "should" not a "must", I don't think we would immediately reject a good-faith contribution simply because it breaks some obscure test. The specific scenario in which a test failure should lead more directly to a quick rejection is where the PR makes some changes to production code and breaks the corresponding unit tests, indicating that the proposed changes need rethinking and that the contributor didn't even take the time run those unit tests to get that feedback for themselves.

Reply:
I like `affected test suites` or `related test suites` better.  

For better or worse, our test suite is still a little fragile (as of this writing, this PR is failing 9 tests in CI, none of which are related).  I routinely review PRs with failing tests that our team members submitted, and while there's a strong expectation that the author will fix them in that case, I don't want us to appear to have a double standard in that regard.

Reply:
Ack, moved to `affected test suites` in 32691a9e628464109d46ada5bf40679e7a2b71d0.